### PR TITLE
Add markdownlint CI check and refactor md files

### PR DESCRIPTION
**Pull request by albertogdd**
## Description

Add [Markdownlint](https://github.com/DavidAnson/markdownlint) GH action on CI.

Fix all existing lint errors in `.md` files.

`.markdownlint.json` config file was already introduced before.

Commands added:

- `make markdown/lint`
- `make prerequisites/markdownlint`

I also set the pre-commit hook to run the linter

## How can this be tested?

Introduce a lint error in a .md file and trigger the action. The PR should get blocked.

You can see I tested that in my commit "Test" of this PR.

## Checklist

~~- [ ] Unit tests have been updated/added~~
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by andriisoldatenko** [comment]
is it possible to add `make md/lint` or something to run linter without VSCode?

File: README.md [hunk 1]
```diff
@@ -42,13 +43,14 @@ objects like permissions, custom resources and corresponding StatefulSets.
 To create the namespace and apply the operator run the following commands
 
 ```sh
-$ kubectl create namespace dynatrace
-$ kubectl apply -f https://github.com/Dynatrace/dynatrace-operator/releases/latest/download/kubernetes.yaml
+kubectl create namespace dynatrace
```
Comment (andriisoldatenko) [inline README.md:46]: BTW it's very good catch by linter, I saw this issue when try to copy this line via the button in markdown copy code.

**Comment by albertogdd** [comment]
@andriisoldatenko 
> is it possible to add `make md/lint` or something to run linter without VSCode?

I just added the commands 
- `make markdown/lint`
- `make prerequisites/markdownlint`

I also set the pre-commit hook to run the linter

**Review by andriisoldatenko — APPROVED** [review]
practically LGTM!

```
➜  dynatrace-operator git:(feature/markdownlint-ci) .git/hooks/pre-commit
Shell: /bin/zsh
Linting markdown files...
markdownlint .
No Go Files to Update
➜  dynatrace-operator git:(feature/markdownlint-ci) gs
On branch feature/markdownlint-ci
Your branch is up to date with 'origin/feature/markdownlint-ci'.

nothing to commit, working tree clean
➜  dynatrace-operator git:(feature/markdownlint-ci) .git/hooks/pre-commit
Shell: /bin/zsh
Linting markdown files...
markdownlint .
README.md:82:1 MD014/commands-show-output Dollar signs used before commands without showing output [Context: "$ kubectl apply -f cr.yaml"]
make: *** [markdown/lint] Error 1
No Go Files to Update
```

**Pull request merged**